src/ProductService.php

```diff
@@ -175,6 +256,7 @@ protected function getAmazonApi() {
    */
   public function getProductData(array $asins, $renew = FALSE) {
     $asins = array_unique($asins);
+    $asins = array_filter($asins);
```

Comment:
just my 2 cents: i think it's nicer if the variable would be assigned only once `$asins = array_unique(array_filter($asins));`
but in the end i don't care :D